cmd/gcp-controller-manager/app/gke_approver.go

```diff
@@ -425,38 +411,30 @@ func validateTPMAttestation(opts GCPConfig, csr *capi.CertificateSigningRequest,
 		glog.Errorf("Failed extracting VM identity from EK certificate: %v", err)
 		return false
 	}
-	hostname := strings.TrimPrefix("system:node:", x509cr.Subject.CommonName)
+	hostname := strings.TrimPrefix(x509cr.Subject.CommonName, "system:node:")
 	if nodeID.Name != hostname {
 		glog.Errorf("VM name in ATTESTATION CERTIFICATE (%q) doesn't match CommonName in x509 CSR (%q)", nodeID.Name, x509cr.Subject.CommonName)
 		return false
 	}
-	if fmt.Sprint(nodeID.ProjectName) != opts.ProjectID {
+	if fmt.Sprint(nodeID.ProjectName) != opts.projectID {
 		glog.Errorf("Received CSR for a different project Name (%d)", nodeID.ProjectName)
 		return false
 	}
 
-	client := oauth2.NewClient(context.Background(), opts.TokenSource)
-	cs, err := compute.New(client)
+	srv := compute.NewInstancesService(opts.compute)
+	inst, err := srv.Get(fmt.Sprint(nodeID.ProjectID), nodeID.Zone, nodeID.Name).Do()
 	if err != nil {
-		glog.Errorf("Creating GCE API client: %v", err)
+		glog.Errorf("Fetching VM data from GCE API: %v", err)
 		return false
 	}
-	srv := compute.NewInstancesService(cs)
-	inst, err := srv.Get(fmt.Sprint(nodeID.ProjectID), nodeID.Zone, nodeID.Name).Do()
+	ok, err = clusterHasInstance(opts, inst.Zone, inst.Id)
 	if err != nil {
-		glog.Errorf("Fetching VM data from GCE API: %v", err)
+		glog.Errorf("Checking VM membership in cluster: %v", err)
 		return false
 	}
-	if validateClusterMembership {
```

Comment:
we should leave this check flag guarded.

Reply:
Done